scripts/build_static_site.sh

```diff
@@ -104,31 +103,25 @@ build() {
     export SOURCECRED_DIRECTORY="${sourcecred_data}"
 
     if [ "${#projects[@]}" -ne 0 ]; then
-        local weightsStr=""
-        if [ -n "${weights}" ]; then
-            weightsStr="--weights ${weights}"
-        fi
         for project in "${projects[@]}"; do
             NODE_PATH="/code/node_modules${NODE_PATH:+:${NODE_PATH}}" \
-                node /code/bin/sourcecred.js load "${project}" $weightsStr
+                node /code/bin/sourcecred.js load "${project}" \
+                ${weights:+--weights "${weights}"}
         done
     fi
 
     if [ "${#project_files[@]}" -ne 0 ]; then
-        local weightsStr=""
-        if [ -n "${weights}" ]; then
-            weightsStr="--weights ${weights}"
-        fi
         for project_file in "${project_files[@]}"; do
             NODE_PATH="/code/node_modules${NODE_PATH:+:${NODE_PATH}}" \
-                node /code/bin/sourcecred.js load --project "${project_file}" $weightsStr
+                node /code/bin/sourcecred.js load --project "${project_file}" \
+                ${weights:+--weights "${weights}"}
         done
     fi
 
     # Do we need to be in sourcecred directory here?
     cd /code
     yarn -s build --output-path "${target}"
-    echo "Output in target ${target}:"
+    printf 'Output in target %s:\n' "${target}"
```

Comment:
Just curious - what are your reasons for having preference for printf over echo?

Reply:
From the commit message:

> Per POSIX, new code [should not use `echo`][echo]. The behavior is
> undefined when the input contains backslashes. This commit fixes `echo`s
> of non-constant strings.
> 
> [echo]: https://pubs.opengroup.org/onlinepubs/009695399/utilities/echo.html#tag_04_41_16

<https://github.com/vsoch/cred-action/pull/4/commits/2c5b595cdb77984fee623be25a746b015dddaafc>

Reply:
Looks like I need to better read your commit messages, they are beautifully documented.